infrastructure/modules/aws/eks/locals.tf

```
@@ -74,22 +74,43 @@ locals {
   }
 
   # List of EKS managed node groups
-  eks_managed_node_groups = merge({
+  eks_managed_node_groups = {
     for key, value in var.eks_managed_node_groups : key => merge(value, {
-      name                       = "${key}-${var.name}",
+      name                       = can(coalesce(value.name)) ? "${try(value.name, "")}-${var.name}" : "${key}-${var.name}",
```

I would say either

```suggestion
      name                       = can(coalesce(value.name)) ? "${value.name}-${var.name}" : "${key}-${var.name}",
```

or
```suggestion
      name                       =try("${value.name}-${var.name}", "${key}-${var.name}"),
```